### PR TITLE
Remain backwards compatible with Python 3.6 and 3.7

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Add ``typing_extensions`` that brings ``TypedDict`` and ``Literal`` to Python<3.8
Modify some regex expressions to be backwards compatible with Python 3.6 and 3.7

cc: @mariusvniekerk

### CODE REVIEW

**Comment by ocefpaf** [comment]
@mariusvniekerk you are way more familiar with the code base here and I don't feel comfortable merging it. But I can issue a new release when you do.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-latest, macos-latest ]
-        python-version: [ 3.8 ]
+        python-version: [ 3.7, 3.8 ]
```
[the remark this answers is not in the document]
Comment (marcelotrevisani) [reply .github/workflows/test.yml:50]: Python 3.6 will fail in one of the platforms to create the env, I think we should drop it

File: conda_lock/invoke_conda.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def _invoke_conda(
     if conda_flags:
         common_args.extend(shlex.split(conda_flags))
 
-    cmd = [str(conda), *command_args, *common_args, *post_args]
+    cmd = list(map(str, [str(conda), *command_args, *common_args, *post_args]))
```
[the remark this answers is not in the document]
Comment (mariusvniekerk) [reply conda_lock/invoke_conda.py:105]: list comprehension plz

File: setup.cfg [hunk 3]
```diff
@@ -40,6 +40,7 @@ install_requires =
     pydantic >=1.8.1
     ruamel.yaml
     poetry
+    typing_extensions
```
Comment (mariusvniekerk) [inline setup.cfg:43]:
```suggestion
    typing-extensions
```

**Review by mariusvniekerk — APPROVED** [review]

**Pull request merged**